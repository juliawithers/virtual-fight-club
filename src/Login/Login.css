input > .login-username {
    width: 100px;
}

input > .login-password {
    width: 100px;
}