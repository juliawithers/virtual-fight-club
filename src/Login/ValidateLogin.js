import React from 'react';
import PropTypes from 'prop-types'

export default function ValidateLogin(props) {
  if (props.message) {
    return (
      <div className="error">{props.message}</div>
    );
  }
  return <></>
}

ValidateLogin.propTypes = {
  message: PropTypes.string
}