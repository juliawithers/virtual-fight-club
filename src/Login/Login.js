import React, { Component } from 'react'
import context from '../context'
import ValidateLogin from './ValidateLogin'

export default class Login extends Component {
    static contextType = context;

    state = {
        username: '',
        password: ''
    }
    // two functions to grab the inputs- change the state

    // at least onChange on both inputs
    handleChangeInput = e => {
        // determine which input was changed and update state
        const value = e.target.value;
        const id = e.target.id;
        if (id === 'login-username') {
            this.setState({
                username: value
            })
        }
        if ( id === 'login-password') {
            this.setState({
                password: value
            })
        }
    }

    // create handler here for submission to call handleLoginSubmit
    handleSubmit = e => {
        // call the one in context and pass the info from state
        e.preventDefault();
        this.context.handleLoginSubmit(this.state.username,this.state.password)
    }

    render(){


        const { username, password } = this.state;

        return(
            <section>
                <h2>Login</h2>
                <form onSubmit={this.handleSubmit}>
                    <div>
                        <label htmlFor="login-username">Username: </label>
                        <input onChange={this.handleChangeInput} type="text" name='login-username' id='login-username' value={username} />
                        <ValidateLogin message={this.context.usernameErr}/>
                    </div>
                    <div>
                        <label htmlFor="login-password">Password: </label>
                        <input onChange={this.handleChangeInput}type="password" name='login-password' id='login-password'
                        value={password} />
                        <ValidateLogin message={this.context.passwordErr}/>
                    </div>
                    <button type='submit'>Login</button>
                </form>
            </section>
        )
    }
}