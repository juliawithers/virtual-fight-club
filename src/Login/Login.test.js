import React from 'react';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom';
import Login from './Login'

import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe(`Login component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Login />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const wrapper = shallow(<Login />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})