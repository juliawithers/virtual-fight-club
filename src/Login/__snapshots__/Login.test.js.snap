// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the UI as expected 1`] = `
<section>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="login-username"
      >
        Username: 
      </label>
      <input
        className="login-username"
        id="login-username"
        name="login-username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="login-password"
      >
        Password: 
      </label>
      <input
        className="login-password"
        id="login-password"
        name="login-password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <ValidateLogin />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;
